### ISSUE TITLE
Maybe point out that `list-dependencies --reverse` only shows units which are active (or enabled, or otherwise referenced by an active unit)

### ISSUE THREAD

**Issue by ghost** [systemd/systemd#9681]
**systemd version the issue has been seen with**
239-5 on Debian testing

**Used distribution**
Debian testing

**Expected behaviour you didn't see**

`systemctl list-dependencies --all --reverse user-$UID.slice` and
`systemctl list-dependencies --all --before user-$UID.slice` are independent of the login status of $UID, except scopes and transient services. Since `user@.service` has `Slice=user-%i.slice`, it has implicit `After=user-%i.slice` and `Requires=user-%i.slice`, and the relation between `user-$UID.slice` and `user@$UID.service` should be printed, if I understand the manpage correctly.

**Unexpected behaviour you saw**

```
# w
 07:07:32 up 2 min,  2 users,  load average: 0.05, 0.04, 0.01
USER     TTY      FROM             LOGIN@   IDLE   JCPU   PCPU WHAT
root     tty1     -                07:05    2:12   0.07s  0.03s -bash
root     pts/0    192.168.1.131    07:06    0.00s  0.02s  0.00s w
```

```
# systemctl list-dependencies --all --reverse user-1000.slice
user-1000.slice
# systemctl list-dependencies --all --before user-1000.slice
user-1000.slice
● └─shutdown.target
```

The reverse depencency from `user-1000.slice` to `user@1000.service` is not printed. On the other hand,

```
# systemctl list-dependencies --reverse -all user-0.slice 
user-0.slice
● ├─session-1.scope
● ├─session-3.scope
● └─user@0.service
# systemctl list-dependencies --before -all user-0.slice
user-0.slice
● ├─session-1.scope
● │ └─shutdown.target
● ├─session-3.scope
● │ └─shutdown.target
● ├─user@0.service
● │ └─shutdown.target
● └─shutdown.target
```

I wonder whether or not the above behavior is related to the incomplete initialization of user-$UID.slice [9512](https://github.com/systemd/systemd/issues/9512) and [9578](https://github.com/systemd/systemd/issues/9578) (if such initialization issues are not my own illusions).

**Steps to reproduce the problem**
See above

**Comment by sourcejedi** [comment]
I think this is due to garbage collection of `user@1000.service`. When the unit is not active, it is removed...

A non-existent unit e.g. `x-y.slice` - `systemctl show x-y.slice` - will give you much the same result.

This is not how I would have *guessed* `list-dependencies` works, but it matches `show`.  Most often it is possible to forget how this works, I think e.g. enabled units are not garbage-collected.

I am not sure why `systemctl list-dependencies` would show information for units which are... synthesized dynamically and do not appear in `systemctl list-units --all`.  I would think it is more useful to show an error, than to pretend that user-1000.slice exists after it has been de-activated.

**Comment by sourcejedi** [comment]
`list-dependencies` behaves the same even for non-existent service units. Slice units also seem to behave a bit differently to service units though.  With service units, `show` output will include `LoadState=not-found`.  Slice units are different, because you can start them without needing any associated unit file... (but you can create one when you want to...)

So, although it seems to me `systemctl list-dependencies x-y.slice` could more helpfully show an error, I don't think it can just be implemented in terms of the current systemd dbus interface.

**Comment by ryutaroh-matsumoto** [comment]
Thanks for the comment. It is also fine if this is considered a feature and this issue is closed. In that case some documentation of this behavior would be helpful for users.

**Comment by sourcejedi** [comment]
Yeah, perfecting documentation is always an issue...

It's implicit in 1) grouping this command under "Unit Commands", 2) the explanation of the first such command `list-units`, and <b>3</b>), the explanation under the `status` command, that systemd loads a unit into memory when you explicitly query it.

Documenting this command more specifically sounds reasonable. I guess the issue is specific issue to `--reverse` / `--after` / `--before`.  It should fine when the specified unit file lists a dependency on another one; the problem is when the *other* unit file lists the dependency.

**Comment by ryutaroh-matsumoto** [comment]
@sourcejedi thanks again for your comment. The documentation of systemd seems generally well-written and almost complete, as almost every question is answered in somewhere in the documentation. But with this particular case I couldn't see what should be expected from `list-dependencies` with `--reverse` or `--before`...

**Comment by poettering** [comment]
Fix waiting in #10569
